CELERY_LOCAL_DEVELOPMENT.md

```diff
@@ -0,0 +1,360 @@
+3. Test it asynchronously: `your_task.delay()`
+4. Monitor execution with Flower or worker logs
+
+### 2. Testing Tasks
+```python
+# In Django shell or tests
+from transcription.tasks import health_check_task
+
+# Test synchronously (good for debugging)
+result = health_check_task.apply()
+print(result.result)
+
+# Test asynchronously
+async_result = health_check_task.delay()
+print(async_result.get(timeout=10))
+```
+
+### 3. Environment-Specific Behavior
+
+The application automatically detects whether Celery is available:
+- If Celery is properly configured and running: tasks run asynchronously
+- If Celery is not available: falls back to synchronous processing
+- Use `use_celery=False` parameter to force synchronous processing
+
+## Troubleshooting
+
+### Common Issues
+
+**1. "Cannot connect to Redis"**
+```bash
+# Check if Redis is running (Mac with Homebrew)
+brew services list | grep redis
+
+# Start Redis if not running
+brew services start redis
+
+# Or check with Docker Compose
+docker-compose ps redis
+
+# Or check system service (Linux)
+sudo systemctl status redis-server
+
+# Check Redis connectivity
+redis-cli ping
+```
+
+**2. "Worker not receiving tasks"**
+```bash
+# Check worker is connected to same Redis instance
+# Verify CELERY_BROKER_URL in settings matches Redis URL
+# Restart worker with correct settings
+```
+
+**3. "Tasks hanging/not completing"**
+```bash
+# Check worker logs for errors
+# Verify task function doesn't have infinite loops
+# Check task time limits in settings
+```
+
+**4. "ImportError: No module named 'celery'"**
+```bash
+# Reinstall dependencies
+pip install -r requirements.txt
+
+# Or install directly
+pip install celery redis
+```
+
+### Logs and Debugging
+
+```bash
+# Verbose worker logging
+celery -A website worker --loglevel=debug
+
+# Check Django logs for task-related errors
+# Enable logging in settings.py:
+LOGGING = {
+    'version': 1,
+    'disable_existing_loggers': False,
+    'handlers': {
+        'file': {
+            'level': 'INFO',
+            'class': 'logging.FileHandler',
+            'filename': 'celery.log',
+        },
+    },
+    'loggers': {
+        'transcription.tasks': {
+            'handlers': ['file'],
+            'level': 'INFO',
+            'propagate': True,
+        },
+    },
+}
+```
+
+## Configuration Reference
+
+### Key Settings in `website/settings.py`
+
+```python
+# Redis connection
+CELERY_BROKER_URL = "redis://localhost:6379/0"
+CELERY_RESULT_BACKEND = "redis://localhost:6379/0"
+
+# Task serialization
+CELERY_ACCEPT_CONTENT = ["json"]
+CELERY_TASK_SERIALIZER = "json"
+CELERY_RESULT_SERIALIZER = "json"
+
+# Task routing
+CELERY_TASK_ROUTES = {
+    "transcription.tasks.*": {"queue": "transcription"},
+}
+
+# Worker configuration
+CELERY_WORKER_PREFETCH_MULTIPLIER = 1
+CELERY_TASK_ACKS_LATE = True
+
+# Time limits (important for long-running transcription tasks)
+CELERY_TASK_SOFT_TIME_LIMIT = 600  # 10 minutes
```

Comment:
Tell me about this. I may be having it transcribe six 2-hr long audio files (chunked into 10-minute segments) in one task. Is 10 minutes a good setting given that?